lib/rspec/core.rb

```diff
@@ -129,6 +129,31 @@ def self.current_example=(example)
     RSpec::Support.thread_local_data[:current_example] = example
   end
 
+  # Set the current scope rspec is executing in
+  # @api private
+  def self.current_scope=(scope)
+    RSpec::Support.thread_local_data[:current_scope] = scope
+  end
+  RSpec.current_scope = :suite
+
+  # Get the current RSpec execution scope
+  #
+  # Returns `:suite` while loading your tests (as soon as RSpec has loaded)
+  #
+  # Returns `:before_example_hook`/`:before_context_hook`/`:before_suite_hook`/
+  # `:after_example_hook`/`:after_context_hook`/`:after_suite_hook` in the respective hooks
+  #
+  # Returns `:before_example_hook`/`:after_example_hook` inside an `around :each` hook,
+  # before and after you call `example.run` respectively.
+  #
+  # Returns `:example` inside it/example blocks
+  #
+  # Returns `nil` after your suite and all hooks are done
```

Comment:
It should return `:suite` afterwards as well

Reply:
Thought a bit about this. Returning `nil` after the suite is finished is an arbitrary quirk that I can't really justify for any reason.

Also If somehow you ran two suites back to back, it would return `:suite` before running the first one, `nil` in-between them (which is after the first and before the second) and `nil` again after the second.

Apologies for the arbitrariness of implementation.

Reply:
What comes to mind is `bisect` and a custom extension that retries failed examples.

Wondering **if** those extensions would need to distinguish pre-suite from post-suite.
Or calling it from a custom formatter. But the formatter would already know if they implement a `stop`/`close` event handler.

I suggest keeping it consistent for now. If it turns out some extension needs this information, we'll figure something out. Returning `:suite` is more elegant, and consistent, and the whole `current_scope` pretty much covers what we needed for `test-prof` and `rspec-rails`.